### DISCUSSION TITLE
HPC running problem with NREL6 problem

### DISCUSSION THREAD

**Discussion by Ryupuriwatt** [mdolab/dafoam#640] [Tutorials]
Hello profressor, now, I am trying to run the case on HPC system and I get below problem for NREL test case. I would like to know have profressor have these error when try to run NREL tutorial and could you please give me suggestion how to fix it. (DAFoam installation in HPC was sucessful)
![e19dad90fabba8f1536161ad60602ae](https://github.com/mdolab/dafoam/assets/115110434/b9681bdc-3a4d-4a6b-9354-11f62658966c)
![ae215f395e34805062f07b9fa36bf03](https://github.com/mdolab/dafoam/assets/115110434/84f483cc-3f1d-4d90-ba15-8888d65d1989)

 I test to run NACA0012 test case using 2 nodes each node is 20 and there is error occur just the optimization iteration exceed

![f03c9f6c468e903d69cb132bcd58f1b](https://github.com/mdolab/dafoam/assets/115110434/4567300e-dda2-4d3f-b18b-ad323c54a33a)
 but when I run this case with 1 node 40 cores optimization success without optimization exceed .
![image](https://github.com/mdolab/dafoam/assets/115110434/b91ebc52-79e5-4a23-b476-5fcb81ba1cde)

the second question that I have is that now I am making a new helicopter rotor blade with 8 million cells so I should use 800 cores with 160G of RAM right? from the website it said that one CPU core and reserve 2 GB memory per 10,000 cells. Is it a good idea to increase Ct(thrust coefficient) and decrease Cmq(moment coefficient) and should I fix the Thrust coefficient(it will help optimizer to find a good solution right?). Third question, from NREL tutorial, I don't see you give a volume constraint so it is not necessary for rotor blade to have volume constraint right?  My last question, from 
DVGeo.addLocalDV("shapey", lower=-1.0, upper=1.0, axis="y", scale=1.0, pointSelect=PS)
I am not really understand meaning of  lower=-1.0, upper=1.0 is it meter. what is the meaning of upper and lower here

**Comment by ryupuriwat** [comment]
And I have a question, profressor said that DAFoam regression test shown some regression tests failed(on HPC) as known issue so If I want to use DARhoSimpleCFoam for transonic wing calculation, can I use it to run simulation or not. ![3dc2fe805ef404c1065f01e53385a05](https://github.com/mdolab/dafoam/assets/109540234/558708e9-7dc9-42a7-9db5-ae59f4a88295)

**Comment by friedenhe** [comment]
I am not sure 100% what caused the seg fault for your case.. I suspect it is related to memory. Please note that you need 100 to 200 GB of memory per million mesh cells. I suggest you start with a small case for testing. Regarding your question about the NACA0012 case, it is completely normal because the optimized drag between the two test cases is the same. I won't be able to answer your question about the objective/constraint function setup... In general, we do want to set up vol constraints; you can check the one used in the ADODG3 tutorial. `lower=-1.0` means the lower bound of the FFD points is -1.0 m. Regarding the regression tests, it is normal. Some tests may fail on HPCs. Your compilation should be correct.

**Comment by Ryupuriwatt** [comment]
Profressor PingHe I would like to ask you when I use many cores on HPC The calculation always stuck at load balancing why this problem is occur. Can DAFoam be able to be used with mesh around 8-11 millions cells? eventhough I use 16 nodes of HPC(450 cores and 2048Gb of RAM) why this problem still occur?
![image](https://github.com/mdolab/dafoam/assets/115110434/7f1ca8c4-b3e9-4c60-8287-c7142f91c460)  and another question if I want to make the objective of optimization better the adjoint accuracy should be higher right? the greatest accuracy for adjoint calculation should be nd (Nested Dissection) right? Because right now when I use IPOPT it can not reach the target residual even in the Mach aero wing tutorial. so to get the lower residual for the optimizer what is the possible way to do it?  below is the optimization iteration which I learned from your video that if inf_pr and inf_du get to lower value the optimization is converge but now I stuck here as shown in the figure.
![image](https://github.com/mdolab/dafoam/assets/115110434/08eb3ebc-43cd-4d1d-94a7-62f8f15c00d1)

**Comment by friedenhe** [comment]
Q1: I am not sure why the seg fault happened, I suggest you start with a lower mesh density and see if it runs. It may not be caused by the mesh size. We did manage to run cases with ~10 million cells without a problem.
Q2: The adjoint accuracy is not impacted by the matrix re-ordering option ND. Can you share the entire log file? I suspect what you saw is normal. We usually expect the optimality to drop by about two orders. Also, you may want to try SLSQP or SNOPT optimizers.
